tools/proteore_topgo/topGO.xml

```diff
@@ -0,0 +1,268 @@
+    #if $background_genes.inputtype.filetype == "file"
+      --background_header="$background_genes.inputtype.header"
+      --background_column="$background_genes.inputtype.column"
+    #end if
+  #end if
+
+    ]]></command>
+
+    <inputs>
+  <conditional name="inputtype">
+    <param name="filetype" type="select" label="Enter your IDs (Ensembl Gene Id)" help="Copy/paste or from a file">
+      <option value="file" selected="true">Input file containing your IDs</option>
+      <option value="copy_paste">Copy/paste your list of IDs</option>
+    </param>
+    <when value="copy_paste">
+      <param name="genelist" type="text" label="Enter a list of IDs">
+        <sanitizer>
+            <valid initial="string.printable">
+                <remove value="&apos;"/>
+            </valid>
+            <mapping initial="none">
+                <add source="&apos;" target="__sq__"/>
+            </mapping>
+        </sanitizer>
+      </param>
+    </when>
+    <when value="file">
+      <param name="genelist" type="data" format="txt,tabular" label="Select your file" help=""/>
+      <param name="column" type="text" label="Column number of IDs" help="For example, fill in 'c1' if it is the first column, 'c2' if it is the second column and so on">
+        <validator type="regex" message="Please enter a column number, for example: 'c1' for the first column">[c]{0,1}[0-9]+</validator>
+      </param>
+      <param name="header" type="boolean" checked="true" truevalue="true" falsevalue="false" label="Does file contain header?" />
+    </when>
+  </conditional>
+  <conditional name="background_genes">
+    <param name="background" type="boolean" checked="false" truevalue="true" falsevalue="false" label="Use your own background (ID list) ?"/>
+    <when value="true">
+      <conditional name="inputtype">
+        <param name="filetype" type="select" label="Enter your background IDs (Ensembl gene IDs)" help="(e.g : ENSG00000139618)">
+          <option value="file" selected="true">Input file containing your background IDs</option>
+          <option value="copy_paste">Copy/paste your background IDs</option>
+        </param>
+        <when value="copy_paste">
+          <param name="genelist" type="text" label="Copy/paste your background IDs" help="IDs must be separated by spaces into the form field, for example: ENSG00000139618 ENSG00000007350">
+            <sanitizer>
+            <valid initial="string.printable">
+                <remove value="&apos;"/>
+            </valid>
+            <mapping initial="none">
+                <add source="&apos;" target="__sq__"/>
+            </mapping>
+            </sanitizer>
+          </param>
+        </when>
+        <when value="file">
+          <param name="genelist" type="data" format="txt,tabular" label="Select file that contains your background IDs list" help=""/>
+          <param name="header" type="boolean" checked="true" truevalue="true" falsevalue="false" label="Does file contain header ?" />
```

Comment:
if you have parameters that are repeater you can also use a macro if you like. 

Does the file contain a header?